audiomentations/augmentations/transforms.py

```diff
@@ -1151,7 +1154,236 @@ def apply(self, samples, sample_rate):
+                " `pip install audiomentations`",
+                file=sys.stderr,
+            )
+            raise
+            
+        assert len(samples.shape) == 1
+        assert samples.dtype == np.float32
+        
+        int_samples = convert_float_samples_to_int16(samples)
+        
+        audio_segment = pydub.AudioSegment(
+            int_samples.tobytes(),
+            frame_rate=sample_rate,
+            sample_width=int_samples.dtype.itemsize,
+            channels=1,
+        )
 
+        audio_segment = pydub.effects.high_pass_filter(
+            audio_segment, self.parameters["cutoff_freq"]
+        )
+        samples = convert_int16_samples_to_float(np.array(audio_segment.get_array_of_samples()))
+        return samples
+
+
+class BandPassFilter(BaseWaveformTransform):
+    """
+    Apply band-pass filtering to the input audio.
+    """
+
+    supports_multichannel = True
+    requires_sample_rate = True
+
+    def __init__(
+        self,
+        min_center_freq=100.0,
+        max_center_freq=1000.0,
+        min_q=1.0,
+        max_q=2.0,
+        p=0.5
+    ):
+        """
+        :param min_center_freq: Minimum center frequency in hertz
+        :param max_center_freq: Maximum center frequency in hertz
+        :param min_q: Minimum ratio of center frequency to bandwidth
+        :param max_q: Maximum ratio of center frequency to bandwidth
+        :param p: The probability of applying this transform
+        """
+        super().__init__(p)
+
+        self.min_center_freq = min_center_freq
+        self.max_center_freq = max_center_freq
+        self.min_q = min_q
+        self.max_q = max_q
+        if self.min_center_freq > self.max_center_freq:
+            raise ValueError("min_center_freq must not be greater than max_center_freq")
+        if self.min_q > self.max_q:
+            raise ValueError("min_q must not be greater than max_q")
+
+    def randomize_parameters(
+        self, samples: np.array, sample_rate: int = None
+    ):
+        """
+        :params samples: (num_channels, num_samples)
+        """
+        super().randomize_parameters(samples, sample_rate)
+
+        self.parameters["center_freq"] = np.random.uniform(
+            low=self.min_center_freq,
+            high=self.max_center_freq
+        )
+        self.parameters["q"] = np.random.uniform(
+            low=self.min_q,
+            high=self.max_q
+        )
+
+    def apply(self, samples: np.array, sample_rate: int = None):
+        try:
+            import pydub
+        except ImportError:
+            print(
+                "Failed to import pydub. Maybe it is not installed? "
+                "To install the optional pydub dependency of audiomentations,"
+                " do `pip install audiomentations[extras]` instead of"
+                " `pip install audiomentations`",
+                file=sys.stderr,
+            )
+            raise
+            
+        assert len(samples.shape) == 1
```

Comment:
This check is not needed here, as the base class (BaseWaveformTransform) already runs an `is_multichannel` check and throws an exception if multichannel audio is given to a mono-only transform

Reply:
Fixed.